src/common/crypto/mbedtls/cryptoprovider.cpp

```diff
@@ -116,6 +117,150 @@ static Error ASN1RemoveTag(const Array<uint8_t>& src, Array<uint8_t>& dst, int t
     return ErrorEnum::eNone;
 }
 
+Error ParseDN(const mbedtls_x509_name& dn, String& result)
+{
+    result.Resize(result.MaxSize());
+
+    int ret = mbedtls_x509_dn_gets(result.Get(), result.Size(), &dn);
+    if (ret <= 0) {
+        return AOS_ERROR_WRAP(ret);
+    }
+
+    result.Resize(ret);
+    return ErrorEnum::eNone;
+}
+
+Error ParsePrivateKey(const String& pemCAKey, mbedtls_pk_context& privKey)
+{
+    mbedtls_ctr_drbg_context ctrDrbg;
+    mbedtls_entropy_context  entropy;
+
+    mbedtls_ctr_drbg_init(&ctrDrbg);
+    mbedtls_entropy_init(&entropy);
+
+    const char* pers = "test";
+
+    int ret = mbedtls_ctr_drbg_seed(&ctrDrbg, mbedtls_entropy_func, &entropy, (const uint8_t*)pers, strlen(pers));
+    if (ret != 0) {
+        return AOS_ERROR_WRAP(ret);
+    }
+
+    ret = mbedtls_pk_parse_key(&privKey, reinterpret_cast<const uint8_t*>(pemCAKey.Get()), pemCAKey.Size() + 1, nullptr,
+        0, mbedtls_ctr_drbg_random, &ctrDrbg);
+    if (ret != 0) {
+        return AOS_ERROR_WRAP(ret);
+    }
+
+    mbedtls_entropy_free(&entropy);
+    mbedtls_ctr_drbg_free(&ctrDrbg);
+
+    return ErrorEnum::eNone;
+}
+
+RetWithError<StaticString<16>> ConvertToString(const Time& time)
```

Comment:
should '16' be some kind of constant in header?

Reply:
It should be public method `ToString` of aos::Time with unit test I guess.

Reply:
done